addons/website_sale/static/src/js/components/media_dialog/website_sale_x2_many_field.js

```diff
@@ -0,0 +1,78 @@
+import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
+import { X2ManyField, x2ManyField } from "@web/views/fields/x2many/x2many_field";
+import { WebsiteSaleMediaDialog } from "./website_sale_media_dialog";
+
+export class WebsiteSaleX2ManyField extends X2ManyField {
+    setup() {
+        super.setup();
+        this.dialogs = useService("dialog");
+        this.orm = useService("orm");
+    }
+
+    addMedia() {
+        this.dialogs.add(WebsiteSaleMediaDialog, {
+            save: (el) => {}, // Simple rebound to fake its execution
+            multiImages: true,
+            customImageSave: this.onCustomImageSave.bind(this),
+            customVideoSave: this.onCustomVideoSave.bind(this),
+        });
+    }
+
+    onCustomVideoSave(videoInfo) {
+        const url = this.getVideoUrl(videoInfo[0].platform, videoInfo[0].videoId, videoInfo[0].params);
+        let video_list = this.props.record.data[this.props.name];
+        video_list.addNewRecord({ position: "bottom" }).then((record) => {
+            record.update({ "name": videoInfo[0].platform + " - [Video]", "video_url": url.href })
+        })
+    }
+
+    onCustomImageSave(attachments) {
+        attachments.forEach(attachment => {
+            this.orm.searchRead("ir.attachment", [["id", "=", attachment.id]], ["datas", "name"], {}).then((result) => {
+                let image_list = this.props.record.data[this.props.name];
+                image_list.addNewRecord({ position: "bottom" }).then((record) => {
+                    record.update({ "name": result[0].name, "image_1920": result[0].datas })
+                });
+            });
+        });        
+    }
```

Comment:
check with @mish-odoo and follow the ES6 guidelines

Reply:
Have made some changes with regards to ES6, also an additional change has been made to reduce the time of rpc calls made for searching records.